policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +69,76 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected. When a TCP probe is
+/// configured there is no path, so we use '/'.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    if let Some(spec) = spec {
+        for container in spec.containers.clone().into_iter() {
+            if let Some(probe) = container.liveness_probe {
+                set_probe_ports(probe, &mut probes, port_names);
+            }
+            if let Some(probe) = container.readiness_probe {
+                set_probe_ports(probe, &mut probes, port_names);
+            }
+            if let Some(probe) = container.startup_probe {
+                set_probe_ports(probe, &mut probes, port_names);
+            }
+        }
+    }
+    probes
+}
+
+fn set_probe_ports(
+    probe: corev1::Probe,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    if let Some(http) = probe.http_get {
+        if let Some(path) = http.path {
```

Comment:
style nit, take it or leave it: could have only a single level of indentation if we wrote:
```suggestion
    if let Some(path) = probe.http_get.and_then(|http| http.path) {
```

Reply:
I originally tried to do this way, but I ran into the issue of still needing to use `http.port` for the later `match`. This way, we lose `http` in the `and_then` and then only have `path`.

Reply:
yeah, never mind, i was wrong about this.